src/metabase/driver/common/parameters/values.clj

```diff
@@ -128,8 +128,10 @@
                                   {:card-id card-id, :tag tag, :type qp.error-type/invalid-parameter})))]
     (try
       (params/map->ReferencedCardQuery
-       (merge {:card-id card-id}
-              (qp/compile (assoc query :parameters params, :info {:card-id card-id}))))
+       (let [query (assoc query :info {:card-id card-id})]
+         (log/tracef "Compiling referenced query for Card %d\n%s" card-id (u/pprint-to-str query))
+         (merge {:card-id card-id}
+                (qp/compile query))))
```

Comment:
This is the actual fix (removing `:parameters params`) but I added extra logging while debugging